pkg/resources/fluentbit/configsecret.go

```diff
@@ -201,6 +202,10 @@ func (r *Reconciler) configSecret() (runtime.Object, reconciler.DesiredState, er
 		}
 	}
 
+	if r.Logging.Spec.FluentbitSpec.FilterKubernetes.BufferSize != "0" && r.Logging.Spec.FluentbitSpec.FilterKubernetes.BufferSize != "" {
```

Comment:
Empty string means that fluentbit will use the default, which is 32k, which is not enough, so in that case I beleive we need to set it to 0 explicitly when we set the defaults.